app/main.py

```diff
@@ -1,3 +1,29 @@
-def cache(func):
-    # Write your code here
-    pass
+import hashlib
+import json
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    cache = {}
+
+    def compute_key(args: tuple, kwargs: dict) -> str:
```

Comment:
1) no additional functions other than the main one should be written inside the decorator
2) You don't need this function to solve this problem